src/hooks/useChartDomain.js

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+
+export default function useChartDomain(data, dataKeys, percentage = 10) {
+  const values = useMemo(() => {
+    if (!data || !data.length || !dataKeys || !dataKeys.length) {
+      return [];
+    }
+    return data.reduce((acc, cv) => {
```

Comment:
sorry, what means cv? Is it "chart value"?

Reply:
It's current value (individual object from the array)

Reply:
1. There is a double validation of parameters here, you can do it at the beginning of the hook

```
if (!data?.length || dataKeys?.length) {
 return []
}
```

2. in one case of invalid params hook returns an empty array, in other case is [null, null], it's a bit strange, maybe we should use some fallback values, like  useChartDomain(..., fallback = [80,, 120])

I don't remember how recharts works and won't everything break if the domain will be [null, null]?

Reply:
1. I don't think we can do at the beginning of the hook because that will lead to the useMemo hook not called and that will result to breaking of the rules of hooks.

2. Yeah makes sense. I have updated to accept a parameter named `fallback` that will be an array.

Reply:
Ah, hooks, yes) forgot that useMemo is used here

But, hm, if this validation returns false, it doesn't mean that there is no NaN or Infinity (Math.min() returns it), for example if this code will be changed, there will be new conditions and domain can be invalid event if data and dataKeys is not empty

Maybe then check something like domain.every(Number.isFinite) && domain[1] >= domain[0] here ?

Reply:
Thanks! Updated